linalg/solvers.cpp

```diff
@@ -146,6 +146,134 @@ void OperatorJacobiSmoother::Mult(const Vector& x, Vector &y) const
    MFEM_FORALL(i, N, Y[i] += X[i] * R[i]; );
 }
 
+OperatorChebyshevSmoother::OperatorChebyshevSmoother(Operator* oper_, const Vector &d,
+                                                     const Array<int>& ess_tdofs,
+                                                     int order_, double max_eig_estimate_)
+   :
+   Solver(d.Size()),
+   order(order_),
+   max_eig_estimate(max_eig_estimate_),
+   N(d.Size()),
+   dinv(N),
+   diag(d),
+   coeffs(order),
+   ess_tdof_list(ess_tdofs),
+   residual(N),
+   oper(oper_) { Setup(); }
+
+void OperatorChebyshevSmoother::Setup()
+{
+   // Invert diagonal
+   residual.UseDevice(true);
+   auto D = diag.Read();
+   auto X = dinv.Write();
+   MFEM_FORALL(i, N, X[i] = 1.0 / D[i]; );
+   auto I = ess_tdof_list.Read();
+   MFEM_FORALL(i, ess_tdof_list.Size(), X[I[i]] = 1.0; );
+
+   // Set up Chebyshev coefficients
+   // See Python script for code generation
```

Comment:
Maybe better to have some kind of reference/citation here?

Reply:
I added a reference to a paper. Is there any place where we can put additional scripts which are useful for MFEM? I wrote a Python script for generating the code for the coefficients.